### PR TITLE
Port to ocidir, drop ociwriter

### PR DESCRIPTION
**Pull request by cgwalters**
Port to ocidir, drop ociwriter

See
https://github.com/ostreedev/ostree-rs-ext/pull/198#issuecomment-1010139066

Basically I messed up a merge.  We want to drop `ociwriter.rs`.
Port all code code to the new `ocidir.rs` which supports derivation
and uses the oci-spec types.

---

### CODE REVIEW

**Comment by cgwalters** [comment]
Depends https://github.com/ostreedev/ostree-rs-ext/pull/206 and https://github.com/ostreedev/ostree-rs-ext/pull/207

**Comment by cgwalters** [comment]
OK rebased :surfer:

File: lib/src/container/encapsulate.rs [hunk 1]
```diff
@@ -71,24 +73,31 @@ fn build_oci(
...
         commit_meta.lookup_value("version", Some(glib::VariantTy::new("s").unwrap()))
     {
         let version = version.str().unwrap();
-        writer.add_config_annotation("version", version);
-        writer.add_manifest_annotation("ostree.version", version);
+        labels.insert("version".into(), version.into());
     }
 
-    writer.add_config_annotation(OSTREE_COMMIT_LABEL, commit);
-    writer.add_manifest_annotation(OSTREE_COMMIT_LABEL, commit);
```
Comment (cgwalters) [inline lib/src/container/encapsulate.rs:83]: I dropped the manifest annotation bit here - it isn't making it into Docker schema, so it's not useful.

File: lib/src/container/ocidir.rs [hunk 2]
```diff
@@ -122,6 +123,7 @@ fn vec_clone_append<T: Clone>(s: &[T], i: T) -> Vec<T> {
 /// Our API right now always mutates a manifest, which means we need
 /// a "valid" manifest, which requires a "valid" config descriptor.
 /// This digest should never actually be used for anything.
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply lib/src/container/ocidir.rs:126]: Good catch!  This was actually leftover from a half done previous port, and actually not using one of these revealed a bug I think.  Looking.

[the remark this answers is not in the document]
Comment (cgwalters) [reply lib/src/container/ocidir.rs]: Yep that was a real bug - and a subtle one https://github.com/containers/oci-spec-rs/issues/88

**Review by jmarrero — APPROVED** [review]
lgtm

**Pull request merged**